src/mappings/gns.ts

```diff
@@ -250,7 +250,10 @@ export function handleSubgraphPublished(event: SubgraphPublished): void {
   // Create subgraph deployment, if needed. Can happen if the deployment has never been staked on
   let subgraphDeploymentID = event.params.subgraphDeploymentID.toHexString()
   let deployment = createOrLoadSubgraphDeployment(subgraphDeploymentID, event.block.timestamp)
-
+  
+  //Associate Contracts and Events with Deployment
+  processManifestForContracts(subgraph,deployment)
```

Comment:
Just curious, is given that we have functions to abstract the notion of getting new metadata from IPFS, what was the reason to directly call this function in the specific event handlers directly, instead of calling it in the function that fetches the metadata? (I'd assume it's a more direct integration, but we would be calling `ipfs.cat` twice, although not sure how much would that affect performance, and I also think it might be because you wanted to avoid having to change the interface of those fetch functions)

Reply:
I'm referring to `fetchSubgraphDeploymentManifest`, which would already fetch the manifest and store it as a String representation of the file in the `SubgraphDeployment` entity btw

Reply:
It was a bit of a design relic from the original subgraph, but seeing as the manifest is only retrieved once per SubgraphDeployment entity, I replaced the IPFS retrieval with processing SubgraphDeployment.manifest and moved the call to createOrLoadSubgraphDeployment.  Let me know if i I should move the call into fetchSubgraphDeploymentManifest instead.